src/selectors/expandThoughts.ts

```diff
@@ -143,9 +146,11 @@ function expandThoughtsRecursive(
         const isAncestor = () => isDescendant(childContext, expansionBasePathContext)
 
         /** Check if the path is equal to the expansion path. */
-        const isExpansionBasePath = () => equalArrays(childContext, expansionBasePathContext)
+        const isExpansionBasePath = equalArrays(childContext, expansionBasePathContext)
+
+        if (!isExpansionBasePath && isPinClosed(child.id)) return false
```

Comment:
Rather than add a separate condition, add it to the final return statement.

Reply:
Sure